### ISSUE TITLE
Undefined reference to `EVP_MD_meth_get_result_size`

### ISSUE THREAD

**Issue by paulmenzel** [hardenedlinux/tpm_pcrtool#1]
On Ubuntu 17.04 with GCC 7.2.0, the build fails with the error below.

```
$ gcc --version | head -1
gcc (Ubuntu 7.2.0-8ubuntu3) 7.2.0
$ make
gcc -Wall   -c -o tpm12.o tpm12.c
gcc -Wall   -c -o tpm2.o tpm2.c
gcc -Wall   -c -o pcrtool.o pcrtool.c
pcrtool.c: In function ‘main’:
pcrtool.c:177:18: warning: format ‘%s’ expects a matching ‘char *’ argument [-Wformat=]
  fprintf(stderr, usagefmt,
                  ^~~~~~~~
pcrtool.c:177:18: warning: format ‘%s’ expects a matching ‘char *’ argument [-Wformat=]
pcrtool.c:177:18: warning: format ‘%s’ expects a matching ‘char *’ argument [-Wformat=]
pcrtool.c:177:18: warning: format ‘%s’ expects a matching ‘char *’ argument [-Wformat=]
pcrtool.c:177:18: warning: format ‘%s’ expects a matching ‘char *’ argument [-Wformat=]
pcrtool.c:177:18: warning: format ‘%s’ expects a matching ‘char *’ argument [-Wformat=]
gcc -Wall   -c -o md.o md.c
md.c: In function ‘MDBIO_md_size’:
md.c:54:10: warning: implicit declaration of function ‘EVP_MD_meth_get_result_size’; did you mean ‘EVP_PKEY_meth_set_derive’? [-Wimplicit-function-declaration]
   return EVP_MD_meth_get_result_size(mdvt);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
          EVP_PKEY_meth_set_derive
gcc -Wall   -c -o fprintpcr.o fprintpcr.c
gcc -o pcrtool tpm12.o tpm2.o pcrtool.o md.o fprintpcr.o -lcrypto -lssl -ltspi -lsapi -ltcti-socket
md.o: In Funktion »MDBIO_md_size«:
md.c:(.text+0xca): Warnung: undefinierter Verweis auf »EVP_MD_meth_get_result_size«
collect2: error: ld returned 1 exit status
Makefile:16: die Regel für Ziel „pcrtool“ scheiterte
make: *** [pcrtool] Fehler 1
```

**Comment by persmule** [comment]
Which version of OpenSSL do you use?
That function should be declared in <openssl/evp.h>, which is included in md.c.

**Comment by persmule** [comment]
My fault. EVP_MD_meth_get_result_size(3ssl) is only available after OpenSSL 1.1, so for compatibility, function EVP_MD_size(3ssl) with a nearly identical effect (both are defined in ${OPENSSL_ROOT}/crypto/evp/evp_lib.c) should be used instead.

**Comment by persmule** [comment]
You could now try the updated version.

**Comment by paulmenzel** [comment]
Thank you for the quick fix. Just for your interest, if you add `Fixes: #1` in the commit message, GitHub closes the issue automatically.
